### PR TITLE
add ant plan

### PR DESCRIPTION
**Pull request by cullenmcdermott**
I saw that #881 already existed, I'm fine with whichever one the maintainers want to take. I'm having trouble installing this package with `hab pkg install` due to the `Crypto error: No public or secret keys found for name_with_rev` so I haven't been able to fully test it but I am able to use it to compile a Cassandra artifact if I build this first.

### CODE REVIEW

**Comment by thesentinels** [comment]
By analyzing the blame information on this pull request, we identified @adamhjk to be a potential reviewer

File: plans/ant/plan.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+pkg_shasum=53dbfb13e0e9bf1b34bcbfdf807c1b9df1b89362b176e1e58150906f518b2c88
+pkg_deps=()
+pkg_build_deps=(core/jdk8)
+pkg_bin_dirs=(bin)
+pkg_include_dirs=(include)
+pkg_lib_dirs=(lib)
+
+do_build() {
+  export JAVA_HOME=$(hab pkg path core/jdk8)
+  ln -s $(hab pkg path core/glibc)/lib/ld-2.22.so /lib/ld-linux-x86-64.so.2
```
Comment (elliott-davis) [inline plans/ant/plan.sh]: this should probably be a patchelf rather than a symlink

Comment (cullenmcdermott) [inline plans/ant/plan.sh]: I've update the PR to use a patchelf instead of a symlink

File: plans/ant/plan.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+pkg_deps=()
+pkg_build_deps=(core/jdk8 core/patchelf)
+pkg_bin_dirs=(bin)
+pkg_include_dirs=(include)
+pkg_lib_dirs=(lib)
+
+do_build() {
+  export JAVA_HOME=$(hab pkg path core/jdk8)
+  export LD_RUN_PATH=$LD_RUN_PATH:$(hab pkg path core/jdk8)/lib/amd64/jli
+  patchelf --interpreter "$(hab pkg path core/glibc)/lib/ld-linux-x86-64.so.2" \
```
Comment (elliott-davis) [inline plans/ant/plan.sh]: This should already be patched as part of jdk8. Is it giving you linking errors?

Comment (elliott-davis) [inline plans/ant/plan.sh]: ah nope - https://github.com/habitat-sh/habitat/blob/master/plans/jdk8/plan.sh#L68
Looks like I missed the 'jre/bin`. I'd love to get that fixed up in the jdk8 plan.

Comment (cullenmcdermott) [inline plans/ant/plan.sh]: #924 should fix this

**Comment by cullenmcdermott** [comment]
It looks like the new version of jdk8 hasn't landed yet so I'm assuming it won't build correctly...yet..

File: plans/ant/plan.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+pkg_origin=core
+pkg_name=ant
+pkg_version=1.9.7
+pkg_maintainer='The Habitat Maintainers <humans@habitat.sh>'
+pkg_license=('Apache 2')
+pkg_source=https://github.com/apache/ant/archive/rel/$pkg_version.tar.gz
+pkg_shasum=53dbfb13e0e9bf1b34bcbfdf807c1b9df1b89362b176e1e58150906f518b2c88
+pkg_deps=()
```
Comment (metadave) [inline plans/ant/plan.sh]: this will need a runtime dep of `core/jdk8`

**Comment by metadave** [comment]
`export JAVA_HOME=$(hab pkg path core/jdk8)`

**Comment by metadave** [comment]
@cullenmcdermott looks good, I was able to start Ant after adding `core/jdk8` to `pkg_deps` and setting `JAVA_HOME`.

it would be great to have a README.md that mentions:

```
# Ant requires JAVA_HOME to be set, and can be set via:
export JAVA_HOME=$(hab pkg path core/jdk8)
```

The same could be added to `core/jdk8`.

**Comment by metadave** [comment]
@cullenmcdermott I added a Maven plan: https://github.com/habitat-sh/habitat/pull/928 which builds with Ant.

File: plans/ant/plan.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+pkg_origin=core
+pkg_name=ant
+pkg_version=1.9.7
+pkg_maintainer='The Habitat Maintainers <humans@habitat.sh>'
+pkg_license=('Apache 2')
```
Comment (metadave) [inline plans/ant/plan.sh]: can you change the license string to read `Apache-2.0`, per the [SPDX License List](http://spdx.org/licenses/)?

**Comment by spuder** [comment]
@metadave How are you starting ant? When I run the pkg, I get the following error. 

`Error: Could not find or load main class org.apache.tools.ant.launch.Launcher
`

```
curl https://raw.githubusercontent.com/cullenmcdermott/habitat-1/451481a5753ecb7b2a1cdadb16c0520201420e70/plans/ant/plan.sh > plan.sh
hab studio -k spuder enter
[0][default:/src:0]# build
...
[2][default:/src:0]# export JAVA_HOME=$(hab pkg path core/jdk8)
[3][default:/src:0]# hab pkg exec spuder/ant ant
/hab/pkgs/spuder/ant/1.9.7/20160621172825/bin/ant: line 85: uname: command not found
/hab/pkgs/spuder/ant/1.9.7/20160621172825/bin/ant: line 102: basename: command not found
/hab/pkgs/spuder/ant/1.9.7/20160621172825/bin/ant: line 115: dirname: command not found
Error: Could not find or load main class org.apache.tools.ant.launch.Launcher
...

**Comment by metadave** [comment]
@spuder my bad, I built via `hab studio build ant`, and then ran it in my Habitat dev shell which happens to have `core/coreutils` installed.

The solution is to change add `core/coreutils` to `pkg_deps`.
        
        pkg_deps=(core/jdk8 core/coreutils)

**Comment by spuder** [comment]
Indeed adding core/coreutils fixes the problem. 

Looking forward to core/ant being released. (metadave/ant still has the coreutils problem).

**Comment by metadave** [comment]
@spuder metadave/ant was just a test build, I don't intend to release artifacts in that origin.

**Comment by cullenmcdermott** [comment]
@spuder @metadave I've pushed a new commit that adds `core/coreutils`

**Comment by metadave** [comment]
@cullenmcdermott great, ready for us to merge?

**Comment by cullenmcdermott** [comment]
Yeah!

**Comment by metadave** [comment]
@thesentinels r+

**Comment by thesentinels** [comment]
:pushpin: Commit d9af99a has been approved by `metadave`

**Comment by thesentinels** [comment]
:hourglass: Testing commit d9af99a with merge d6b4aed...

**Comment by metadave** [comment]
thanks @cullenmcdermott 

built and pushed to the depot: https://app.habitat.sh/#/pkgs/core/ant/1.9.7/20160621210022

**Comment by thesentinels** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/habitat-sh/habitat/builds/139325177)

**Pull request merged**